handbook/engineering/cloud/index.md

```diff
@@ -6,6 +6,26 @@ The cloud team owns all work that is necessary to build, secure, scale, and oper
 
 1. Any user or organization can use Sourcegraph Cloud with both the private and public code that they care about in a secure way.
 
+## Milestones
+
+1. Any user can add public code
+2. Organizations can use Sourcegraph with up to 50 repos or 100k LoC of private repositories for free
+    - We need to build a scalable and reliable service to count lines of code and integrate that with the rest of our systems (to enforce limits). This would align with how we’re thinking about billing by LoC, not number of repositories.
```

Comment:
This is super tricky. I know we have had an RFC about this before, and I can't remember exactly all the different things we discussed. Something that occurred to me is when measuring LoC, which branch are we measuring? What if its a repo with lots of very different branches? What if someone is naughty and has a mostly empty HEAD, but does dev on another branch. How do we capture the value of our tool which works across history and branches?

I proposed in a call now rather than measuring LoC, we can measure the size of the `.git` repos. This gets around a lot of these issues, but does have concerns around fluctating sizes due to repacking (but order of magnitudes shouldn't change) and it will measure checked in binaries/deps.

Another idea is to measure the size of our indexed search shards. That will exclude binaries and more directly is related to our costs of serving a search index. However, that is very specific to indexed search and is hard to keep consistent as we explore different technologies/products/etc.

Reply:
Added a sub-item under "Billing and subscriptions" to measure value by size of repositories.

Reply:
> This is super tricky. I know we have had an RFC about this before, and I can't remember exactly all the different things we discussed. Something that occurred to me is when measuring LoC, which branch are we measuring? What if its a repo with lots of very different branches? What if someone is naughty and has a mostly empty HEAD, but does dev on another branch. How do we capture the value of our tool which works across history and branches?
> 
> I proposed in a call now rather than measuring LoC, we can measure the size of the `.git` repos. This gets around a lot of these issues, but does have concerns around fluctating sizes due to repacking (but order of magnitudes shouldn't change) and it will measure checked in binaries/deps.
> 
> Another idea is to measure the size of our indexed search shards. That will exclude binaries and more directly is related to our costs of serving a search index. However, that is very specific to indexed search and is hard to keep consistent as we explore different technologies/products/etc.

I think LOC is a bad measurement / tough sell, but I really like per user. In theory users should relate to lines of code over time, and it nicely avoids the multi-repo/monorepo problem whereby we have to guess the number of repositories to bill appropriately.  Similarly I'd like suggest not measuring the size of .git as it hard limits us to a specific revision control system; one can easily find Team Foundation Server (TFS), and Perforce still going reasonably strong internally.

Reply:
This is a great discussion to have in a forum where the sales team can be involved FYI! I have some thoughts/responses to all of the questions above, but I'll wait to do it in a different venue.

Reply:
This was initially discussed in [RFC 146](https://docs.google.com/document/d/1H1yGMVAgAUwSYHJL-94Glx8JFelx8mnoEhZdS0WOhgc/edit#heading=h.trqab8y0kufp)